### PR TITLE
Safely update or delete records using transactions!

### PR DESCRIPTION
**Pull request by iAmWaris97**
In this milestone I have done the following:

- Inside a transaction:

- I have added a new column to the animals table.
- I have inserted some new data into the animals table.

- [ ] Using the transaction:

- _Inside a transaction update the animals table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction._

- [ ] Using the transaction:

- _Update the animals table by setting the species column to digimon for all animals that have a name ending on mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Commit the transaction._

- [ ] Using the transaction:

- _Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exist._

 [ ] Using the transaction:

- _Delete all animals born after Jan 1st, 2022.

- Create a savepoint for the transaction.
- Update all animals' weights to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction

Write queries to answer the following 

- [ ] questions:

_- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?_

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,3 +5,10 @@ INSERT INTO animals VALUES (1,'Agumon', '02-03-2020',0,true,10.23);
 INSERT INTO animals VALUES (2,'Gabumon', '11-15-2018',2,true,8);    
 INSERT INTO animals VALUES (3,'Pikachu', '01-07-2021',1,false,15.04);
 INSERT INTO animals VALUES (4,'Devimon', '05-12-2017',5,true,11);
+INSERT INTO animals VALUES (5,'Charmander', '02-08-2020',0,false,11);
+INSERT INTO animals VALUES (6,'Plantmon', '11-15-2021',2,true,5.7);
+INSERT INTO animals VALUES (7,'Squirtle', '04-02-1993',3,false,12.13);
+INSERT INTO animals VALUES (8,'Angemon', '06-12-2015',1,true,45);
+INSERT INTO animals VALUES (9,'Boarmon', '06-07-2005',7,true,20.4);
+INSERT INTO animals VALUES (10,'Blossom', '10-13-1998',3,true,17);
+INSERT INTO animals VALUES (11,'Ditto', '05-14-2022',4,true,22);
```
Comment (BenMukebo) [inline data.sql:14]: - [ ] When storing a date value, PostgreSQL uses the `yyyy-mm-dd format` e.g. 1994-10-27. This format is also used in PostgreSQL for inserting data. Please let's modify this 👍🏻 
For more documentation visit this l[ink ](https://blog.devart.com/postgres-date-types-and-functions-formatting.html)


- [ ] Kindly let's be more careful. According to the Project requirements, you should Insert the following data 👇🏻  As you can see the fourth first data has a negative `weight_kg` value, kindly let's respect this 💯 👍🏻  

> - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
> - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
> - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
> - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to

- This should look like this 👇🏻 

```sql
   INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Charmander', '2020/2/8', '0', 'false', '-11.0');

    INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Plantmon', '2021/11/15', '2', 'true', '-5.7');

    INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Squirtle', '1993/04/02', '3', 'false', '-12.13');

    INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Angemon', '2005/06/12', '1', 'true', '-45.0');
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @iAmWaris97,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (iAmWaris97) [reply data.sql:14]: Thank you for the kind review, I have updated the data as requested and saved the queries into the queries.sql.
Following snapshot shows the data available inside the animals table now:

chanages

**Review by alexander16108 — APPROVED** [review]
Hi @iAmWaris97 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iAmWaris97** [comment]
Hi @alexander16108,

Thank you for your review and approval, it is so kind of you.

**Pull request merged**